cdap-docs/build.sh

```diff
@@ -460,18 +497,128 @@ function copy_docs_inner_level() {
 }
 
 function build_zip() {
+  local doc_target=${1}
+  local zip_target=${2}
+  local zip_extras=${3}
   set_project_path
-  make_zip ${1}
-}
-
-function zip_extras() {
-  if [[ "x${1}" == "x" ]]; then
-    return
+  set_version
+  local zip_dir_name
+  if [[ -n ${zip_target} ]]; then
+    zip_dir_name="${PROJECT}-docs-${PROJECT_VERSION}"
+  else
+    zip_dir_name="${PROJECT}-docs-${PROJECT_VERSION}-$1"
+  fi
+  local zip_dir_name_archive
+  local build_flag
+  if [ "x${doc_target}" == "x${BUILD_GITHUB}" ]; then
+    zip_dir_name_archive="${zip_dir_name}-github-archive.zip"
+    build_flag="--ghpages"
+  else
+    zip_dir_name_archive="${zip_dir_name}-web-archive.zip"
   fi
-  echo "Adding .htaccess file (404 file)"
-  rewrite ${SCRIPT_PATH}/${COMMON_SOURCE}/${HTACCESS} ${TARGET_PATH}/${PROJECT_VERSION}/.${HTACCESS} "<version>" "${PROJECT_VERSION}"
   cd ${TARGET_PATH}
-  zip -qr ${ZIP_DIR_NAME}.zip ${PROJECT_VERSION}/.${HTACCESS}
+  doc_change_py="${TARGET_PATH}/../tools/doc-change.py"
+
+  echo "Removing old directories and zips"
+  rm -rf base current ${PROJECT_VERSION} *.zip
+  
+  echo "Creating base"
+  mkdir base
+  cp -r ${HTML} base/en
+  errors=$?
+  if [[ ${errors} -ne 0 ]]; then
+      echo "Could not create base"
+      return ${errors}   
+  fi
+
+  # Add a redirect index.html file
+  cp ${SCRIPT_PATH}/${COMMON_SOURCE}/redirect.html base/index.html
+  errors=$?
+  if [[ ${errors} -ne 0 ]]; then
+      echo "Could not add redirect file"
+      return ${errors}   
+  fi
+
+  echo
+  echo "1: 'current' version, with canonical ref link ${zip_dir_name}-current"
+  cp -r base current
+  python ${doc_change_py} ${build_flag} --type=current --version=${PROJECT_VERSION} ${TARGET_PATH}/current
+  errors=$?
+  if [[ ${errors} -ne 0 ]]; then
+      echo "Could not change doc set ${TARGET_PATH}/current"
+      return ${errors}   
+  fi
+  zip -qr ${zip_dir_name}.zip current/* --exclude *.DS_Store* *.buildinfo*
+  if [[ "x${zip_extras}" != "x" ]]; then
+    echo "Creating and adding a .htaccess file (404 file)"
+    rewrite ${SCRIPT_PATH}/${COMMON_SOURCE}/${HTACCESS} ${TARGET_PATH}/current/.${HTACCESS} "<version>" "current"
+    zip -qr ${zip_dir_name}.zip current/.${HTACCESS}
+  fi
+  mv ${zip_dir_name}.zip ${zip_dir_name}-current.zip
+  errors=$?
+  if [[ ${errors} -ne 0 ]]; then
+      echo "Could not move ${zip_dir_name}.zip to ${zip_dir_name}-current.zip"
+      return ${errors}   
+  fi
```

Comment:
Even better, why don't we just create `${zip_dir_name}-current.zip` and not bother with the `mv`? Seems these three sections could be refactored into a single function and have less maintenance in the long run, as it appears to be 3 slight variations on the same steps.